src/components/calculator.jsx

```diff
@@ -1,26 +1,53 @@
+import { useState } from 'react';
+import calculate from '../logic/calculate';
 import Buttons from './buttons';
 import './calculator.scss';
 
 // Function to create each button of the calculator
-const createButtons = () => {
+const createButtons = (changeResult) => {
   const buttonsInput = ['AC', '+/-', '%', '÷', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
   const buttons = [];
   buttonsInput.forEach((value) => {
-    buttons.push(<Buttons value={value} />);
+    buttons.push(
+      <Buttons
+        key={value}
+        value={value}
+        setResult={changeResult}
+        calculate={calculate}
+      />,
+    );
   });
 
   return buttons;
 };
```

Comment:
- [ ] @NicolasAndrehh great work on your components however it's not the best practice to have two components in the same file, kindly create a file for the createButtons component and then import it into the calculator component.